test/integration/agent/gobgp_test.go

```diff
@@ -0,0 +1,349 @@
+	asn1 := int32(61179)
+	asn2 := int32(62179)
+	asn3 := int32(63179)
+	routerID1 := "192.168.1.1"
+	routerID2 := "192.168.1.2"
+	routerID3 := "192.168.1.3"
+	listenPort1 := int32(1179)
+	listenPort2 := int32(2179)
+	listenPort3 := int32(3179)
+	server1GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn1),
+		RouterID:   routerID1,
+		ListenPort: listenPort1,
+	}
+	server2GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn2),
+		RouterID:   routerID2,
+		ListenPort: listenPort2,
+	}
+	server3GlobalConfig := &bgp.GlobalConfig{
+		ASN:        uint32(asn3),
+		RouterID:   routerID3,
+		ListenPort: listenPort3,
+	}
+
+	server1 := gobgp.NewGoBGPServer(server1GlobalConfig)
+	server2 := gobgp.NewGoBGPServer(server2GlobalConfig)
+	server3 := gobgp.NewGoBGPServer(server3GlobalConfig)
+
+	ctx := context.Background()
+	t.Log("Starting all BGP servers")
+	require.NoError(t, server1.Start(ctx))
+	require.NoError(t, server2.Start(ctx))
+	require.NoError(t, server3.Start(ctx))
+	t.Log("Started all BGP servers")
+
+	ipv4Server1Config := bgp.PeerConfig{
+		BGPPeer: &v1alpha1.BGPPeer{
+			Address:                    "127.0.0.1",
+			Port:                       ptr.To[int32](1179),
+			ASN:                        61179,
+			MultihopTTL:                ptr.To[int32](2),
+			GracefulRestartTimeSeconds: ptr.To[int32](120),
+		},
+	}
+	ipv6Server1Config := bgp.PeerConfig{
+		BGPPeer: &v1alpha1.BGPPeer{
+			Address:                    "::1",
+			Port:                       ptr.To[int32](1179),
+			ASN:                        61179,
+			MultihopTTL:                ptr.To[int32](2),
+			GracefulRestartTimeSeconds: ptr.To[int32](120),
+		},
+	}
+	ipv4Server2Config := bgp.PeerConfig{
+		BGPPeer: &v1alpha1.BGPPeer{
+			Address:                    "127.0.0.1",
+			Port:                       ptr.To[int32](2179),
+			ASN:                        62179,
+			MultihopTTL:                ptr.To[int32](3),
+			GracefulRestartTimeSeconds: ptr.To[int32](130),
+		},
+	}
+	ipv6Server3Config := bgp.PeerConfig{
+		BGPPeer: &v1alpha1.BGPPeer{
+			Address:                    "::1",
+			Port:                       ptr.To[int32](3179),
+			ASN:                        63179,
+			MultihopTTL:                ptr.To[int32](1),
+			GracefulRestartTimeSeconds: ptr.To[int32](140),
+		},
+	}
+
+	t.Log("Adding BGP peers for BGP server1")
+	require.NoError(t, server1.AddPeer(ctx, ipv4Server2Config))
+	require.NoError(t, server1.AddPeer(ctx, ipv6Server3Config))
+	t.Log("Added BGP peers for BGP server1")
+
+	t.Log("Adding BGP peers for BGP server2")
+	require.NoError(t, server2.AddPeer(ctx, ipv4Server1Config))
+	t.Log("Added BGP peers for BGP server2")
+
+	t.Log("Adding BGP peers for BGP server3")
+	require.NoError(t, server3.AddPeer(ctx, ipv6Server1Config))
+	t.Log("Added BGP peers for BGP server3")
+
+	getPeersFn := func(server bgp.Interface) sets.Set[string] {
+		peerKeys := sets.New[string]()
+		peers, err := server.GetPeers(ctx)
+		if err != nil {
+			return peerKeys
```

Comment:
nit: maybe returning `nil` would be better here

Reply:
Done